File: src/main/java/com/autowashpro/service/VehicleService.java
package com.autowashpro.service;

import com.autowashpro.dto.request.VehicleCreateRequest;
import com.autowashpro.dto.request.VehicleStatusUpdateRequest;
import com.autowashpro.dto.request.VehicleUpdateRequest;
import com.autowashpro.dto.response.PageResponse;
import com.autowashpro.dto.response.VehicleResponse;

import java.util.List;

public interface VehicleService {
    VehicleResponse create(VehicleCreateRequest request, Long customerId);
    List<VehicleResponse> listOwn(Long customerId);
    VehicleResponse getOwnById(Long id, Long customerId);
    VehicleResponse update(Long id, VehicleUpdateRequest request, Long customerId);
    VehicleResponse setDefault(Long id, Long customerId);
    VehicleResponse updateStatus(Long id, VehicleStatusUpdateRequest request, Long customerId);
    PageResponse<VehicleResponse> adminList(int page, int limit, String vehicleType, String keyword);

    /**
     * Dùng cho Booking API — ném 400 nếu vehicle không active
     */
    void assertVehicleIsActive(Long vehicleId);
}

File: frontend/src/services/vehicleService.js
import api from './api'

export const vehicleService = {
  async listOwn() {
    const res = await api.get('/api/vehicles')
    return res.data
  },

  async create(payload) {
    const res = await api.post('/api/vehicles', payload)
    return res.data
  },

  async update(id, payload) {
    const res = await api.patch(`/api/vehicles/${id}`, payload)
    return res.data
  },

  async setDefault(id) {
    const res = await api.patch(`/api/vehicles/${id}/default`)
    return res.data
  },

  async updateStatus(id, isActive) {
    const res = await api.patch(`/api/vehicles/${id}/status`, { isActive })
    return res.data
  },

  async adminList(params = {}) {
    const res = await api.get('/api/admin/vehicles', { params })
    return res.data
  },
}
